lib/thesis/models/file.ex

```diff
@@ -35,11 +35,14 @@ defmodule Thesis.File do
     file
     |> cast(%{content_type: f.content_type}, [:content_type])
     |> cast(%{filename: f.filename}, [:filename])
-    |> cast(%{data: File.read!(f.path)}, [:data])
+    |> cast(%{data: get_file_data(f.path)}, [:data])
     |> cast(%{slug: generate_slug(f.filename)}, [:slug])
     |> validate_required(@required_attributes)
   end
 
+  defp get_file_data(%{data: data}), do: data
+  defp get_file_data(path), do: File.read!(path)
+
```

Comment:
since we are passing in the actual file binary instead of the file path, a function here to differentiate between he two. the standard upload plug will always pass in a path while we are knowingly passing in the data in a map during the import.